elastic/apm-server
# Add terraform module to create soaktest workers on GCP

## Motivation/summary

Workers will generate continuous and sustained load on the APM-Server to allow for soak testing. Workers will be long running processes with capabilities to recover automatically on failures or generate alert notifications for non-recoverable errors. The terraform script included in this PR makes managing workers easy.

## Checklist

~~- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)~~
~~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~~

## How to test these changes

As part of the soak test project, the worker should already be running and sending events to a central ESS APM-Server. If that is not the case, the terraform module can be tested by:

```
$ cd testing/apmsoak
$ make apply # this will ask for some configurations related to the target APM Server
```

On success, a worker node should be created sending data to the target APM Server. Assert that APM Server is receiving load.

## Related issues

Closes #8534
